internal/services/search/search_service_resource.go

```diff
@@ -64,21 +67,76 @@ func resourceSearchService() *pluginsdk.Resource {
 					string(services.SkuNameStandardTwo),
 					string(services.SkuNameStandardThree),
 					string(services.SkuNameStorageOptimizedLOne),
-					string(services.SkuNameStorageOptimizedLOne),
+					string(services.SkuNameStorageOptimizedLTwo),
 				}, false),
 			},
 
 			"replica_count": {
-				Type:     pluginsdk.TypeInt,
-				Optional: true,
-				Computed: true,
+				Type:         pluginsdk.TypeInt,
+				Optional:     true,
+				Default:      1,
+				ValidateFunc: validation.IntBetween(1, 12),
 			},
 
 			"partition_count": {
 				Type:         pluginsdk.TypeInt,
 				Optional:     true,
-				Computed:     true,
-				ValidateFunc: validation.IntAtMost(12),
+				Default:      1,
+				ValidateFunc: validateSearch.PartitionCount,
+			},
+
+			"api_access_control": {
+				Type:     pluginsdk.TypeSet,
+				Optional: true,
+				Computed: true,
+				MaxItems: 1,
+				Elem: &pluginsdk.Resource{
+					Schema: map[string]*pluginsdk.Schema{
+						"type": {
+							Type:     pluginsdk.TypeString,
+							Optional: true,
+							Default:  "api_keys",
```

Comment:
I agree that this would work and have made the change in my code, however I will have to respectfully disagree with the naming of the `local_authentication_enabled` field. I feel that we should match the API here and name the field `local_authentication_disabled`. My reasoning here is that we have seen the confusion which resulted from us flipping the meaning of these sorts of values in the past when compared to the public documentation as we did with the `Private Link Network Rules` field in the `Subnet` resources.